CodePush.js

```diff
@@ -412,10 +413,14 @@ if (NativeCodePush) {
     restartApp,
     setUpTestDependencies,
     sync,
+    disallowRestart: RestartManager.disallow,
+    allowRestart: RestartManager.allow,
+    restartAllowed: RestartManager.allowed,
     InstallMode: {
       IMMEDIATE: NativeCodePush.codePushInstallModeImmediate, // Restart the app immediately
       ON_NEXT_RESTART: NativeCodePush.codePushInstallModeOnNextRestart, // Don't artificially restart the app. Allow the update to be "picked up" on the next app restart
-      ON_NEXT_RESUME: NativeCodePush.codePushInstallModeOnNextResume // Restart the app the next time it is resumed from the background
+      ON_NEXT_RESUME: NativeCodePush.codePushInstallModeOnNextResume, // Restart the app the next time it is resumed from the background
+      ON_NEXT_RESTART_OPPORTUNITY: NativeCodePush.codePushInstallModeOnNextRestartOpportunity
```

Comment:
Do we actually need the new install mode for this? The install mode represents when the update should be attempted to be installed automatically, but it may happen sooner (e.g. if the app calls `restartApp`), and it could also happen later (e.g. if you specified a `minimumBackgroundDurarion`). The `ON_NEXT_RESTART` install mode basically disables automatically forcing a restart, so the name of this new install mode is also potentially a little confusing, since it's behavior is more closely aligned with `IMMEDIATE`.

I would vote to remove this install mode, since I think this new behavior is conceptually similar to what `minimumBackgroundDuration` already does for resume-based installs, and therefore, it seems fine to have a way to delay immediate installs without requiring a new install mode.

As an aside, should `restartApp` even be allowed to be called if you've explicitly disallowed restarts? It feels like `restartApp` is really the "primitive" that allows restarting the app, and if we're adding the ability to disallow it, then maybe that's where the delay behavior should go. Then, anywhere that calls `restartApp` would simply be allowed or not, but they wouldn't need to worry about interacting with the `RestartManager`.